src/ExceptionAll.Client/Exceptions/ExceptionAllException.cs

```diff
@@ -0,0 +1,23 @@
+﻿namespace ExceptionAll.Client.Exceptions;
+
+public class ExceptionAllException : Exception
+{
+    public HttpResponseMessage ResponseMessage { get; private set; }
+    public ApiErrorDetails ErrorDetails { get; private set; }
+
+    public ExceptionAllException(HttpResponseMessage responseMessage, ApiErrorDetails errorDetails) 
+        : base(errorDetails.Message)
+    {
+        ResponseMessage = responseMessage;
+        ErrorDetails    = errorDetails;
+    }
+
+    public ExceptionAllException(
```

Comment:
Kinda weird to have an "optional" param be at the beginning here - since we already have `ExceptionAllException(HttpResponseMessage responseMessage, ApiErrorDetails errorDetails)`.

I think innerException should go at the end of this parameter list